### ISSUE TITLE
Provide a registry API for supported elements/media types

### ISSUE THREAD

**Issue by ocean90** [rqrauhvmra/Tobii#42]
By default the library supports images, inline HTML, iframes and YouTube videos. There's already a comment "[you can add new type to support something new](https://github.com/midzer/tobii/blob/7804fa093412e5155eebed508c54e4dae0257836/src/js/tobii.js#L115)" but that's not true unless you change the source directly.
Idea: Provide an API that allows to add and remove media types.

This is related to #1 to split the library into modules. I think it would be nice if each media type is its own module which can be added manually. This should reduce the package size for users who want to use the library only for images.
A possible implementation may look like this:

```js
// import Tobii from 'tobii'; // This would continue to include all types
import { Base as Tobii } from 'tobii'; // This one requires you to manually register the types
import image from 'tobii/types/image';
import youtube from 'tobii/types/youtube';
import myCustomType from './my-custom-type';

Tobii.registerType(image);
Tobii.registerType(youtube);
Tobii.registerType(myCustomType);

new Tobii();
```

[Inspired by the Day.js library](https://day.js.org/docs/en/plugin/relative-time).

What are your thoughts on this idea? Are there better implementation ideas?

**Comment by midzer** [comment]
I really like the idea! The possibility to have smaller bundle sizes is always good.

I'm kind of stuck at https://github.com/midzer/tobii/issues/1 and any help is warmly appreciated.

**Comment by viliusle** [comment]
Maybe it is too old ticket to reply, but do we really want to reduce file size, if total library size is only 8.6 kb compressed? It is already small.

p.s. I checked other 2 similar libs, results were 22kb and 38kb.

**Comment by midzer** [comment]
@viliusle You're right. I think we have no urgency. But if some1 is willing to jump on this item, it would be nice for a tobii v3.

**Comment by midzer** [comment]
@ocean90 Would you like to work in this for the upcoming v3?
